File: fixtures/scale/r/small/lib/border.r
# Box-drawing borders around rendered blocks.

border_rule <- function(width, ch = "-") {
  paste0("+", strrep(ch, width), "+")
}

border_wrap <- function(lines, width) {
  boxed <- sprintf("|%s|", formatC(lines, width = -width))
  c(border_rule(width), boxed, border_rule(width))
}

File: fixtures/scale/julia/small/src/Border.jl
module Border

export border_wrap

function border_rule(width::Int, ch::String = "-")
    return "+" * repeat(ch, width) * "+"
end

function border_wrap(lines::Vector{String}, width::Int)
    boxed = ["|" * rpad(line, width) * "|" for line in lines]
    return vcat([border_rule(width)], boxed, [border_rule(width)])
end

end
